clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -71,8 +113,26 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals, boolean doLog)
             logAll();
     }
 
+    /**
+     * Construct a configuration with a ConfigDef, the configuration properties, optional {ConfigProviders}.
+     *
+     * @param definition the definition of the configurations; may not be null
+     * @param originals of config provider configs as well as configuration properties; may not be null
+     */
     public AbstractConfig(ConfigDef definition, Map<?, ?> originals) {
-        this(definition, originals, true);
+        this(definition, originals, Collections.emptyMap(), false);
+    }
+
+    /**
+     * Construct a configuration with a ConfigDef, the configuration properties, optional {ConfigProviders}.
```

Comment:
This constructor doesn't take "optional {ConfigProviders}", does it? BTW, the `{ConfigProviders}` is not correct; maybe `{@link ConfigProviders}`? Same comment for line 117.

I'd suggest something like:
```
Construct a configuration with a ConfigDef and the configuration properties, 
which can include properties for zero or more {@link ConfigProviders} 
that will be used to resolve variables in configuration property values.
```

Reply:
Sorry thats a left over fixed it